test/testdata/lsp/code_actions/extract_variable/other.rb

```diff
@@ -0,0 +1,24 @@
+# typed: true
+# selective-apply-code-action: refactor.extract
+
+if 1 == 2
+  1 + 1
+#     ^ apply-code-action: [A] Extract Variable
```

Comment:
Out of curiosity, did you look at any other languages to see whether/how they decide to handle this case?